### PR TITLE
Separate SqlServer testBasicTypes into multiple tests

### PR DESCRIPTION
**Pull request by tangjiangling**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

SqlServer type mapping tests.

> How would you describe this change to a non-technical end user or system administrator?

N/A.

## Related issues, pull requests, and links

Related #11181

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 1]
```diff
@@ -89,21 +89,96 @@ public void setUp()
...
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_smallint"));
+    }
+
+    @Test
+    public void testUnsupportedSmallint()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_unsupported_smallint", "(data smallint)")) {
+            assertQueryFails(
+                    format("INSERT INTO %s VALUES (-32769)", table.getName()), // min - 1
+                    "Out of range for smallint: -32769");
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:140]: This error comes from Trino engine (not the connector nor SQL Server). Please execute the query on SQL Server to verify the actual limitation in the remote database.

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 2]
```diff
@@ -89,21 +89,96 @@ public void setUp()
...
+    @Test
+    public void testTinyint()
+    {
+        // TODO: Add min/max/min-1/max+1 tests (https://github.com/trinodb/trino/issues/11209)
+        SqlDataTypeTest.create()
+                .addRoundTrip("tinyint", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
                 .addRoundTrip("tinyint", "5", TINYINT, "TINYINT '5'")
-                .execute(getQueryRunner(), trinoCreateAsSelect("test_basic_types"));
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_tinyint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_tinyint"));
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:119]: Please execute with `trinoCreateAndInsert` as well. Subsequent test are also.

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java (hunk 1 above)
Comment (tangjiangling) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java]: Updated.

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 3]
```diff
@@ -89,21 +94,101 @@ public void setUp()
...
+    public void testBigint()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("bigint", "NULL", BIGINT, "CAST(NULL AS BIGINT)")
+                .addRoundTrip("bigint", "-9223372036854775808", BIGINT, "-9223372036854775808") // min value in SQL Server and Trino
+                .addRoundTrip("bigint", "123456789012", BIGINT, "123456789012")
+                .addRoundTrip("bigint", "9223372036854775807", BIGINT, "9223372036854775807") // max value in SQL Server and Trino
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_bigint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_bigint"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_bigint"));
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:191]: Let's add `testUnsupportedBigint` as well.

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 4]
```diff
@@ -89,21 +94,101 @@ public void setUp()
...
+                .addRoundTrip("smallint", "32767", SMALLINT, "SMALLINT '32767'") // max value in SQL Server and Trino
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_smallint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_smallint"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_smallint"));
+    }
+
+    @Test
+    public void testUnsupportedSmallint()
+    {
+        try (TestTable table = new TestTable(sqlServer::execute, "test_unsupported_smallint", "(data smallint)")) {
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:145]:
```suggestion
        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_smallint", "(data smallint)")) {
```

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 5]
```diff
@@ -89,21 +94,101 @@ public void setUp()
...
+                .addRoundTrip("integer", "2147483647", INTEGER, "2147483647") // max value in SQL Server and Trino
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_int"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_int"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_int"));
+    }
+
+    @Test
+    public void testUnsupportedInteger()
+    {
+        try (TestTable table = new TestTable(sqlServer::execute, "test_unsupported_integer", "(data integer)")) {
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:171]:
```suggestion
        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_integer", "(data integer)")) {
```

**Review by ebyhr — APPROVED** [review]
Looks good except for comments. Please check CI failure about "undeclared dependencies". 

Also, you can squash the 3rd commit into 2nd commit wth the below commit message:
```
Improve SQL Server type mapping tests
```

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 6]
```diff
@@ -762,4 +857,24 @@ protected SqlExecutor onRemoteDatabase()
...
+    private void assertSqlServerQueryFails(@Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp)
+    {
+        try {
+            sqlServer.execute(sql);
+            fail(format("Expected query to fail: %s", sql));
+        }
+        catch (RuntimeException exception) {
+            assertThat(getSqlServerExceptionCause(exception)).hasMessageMatching(expectedMessageRegExp);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (tangjiangling) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java]: Updated.

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 7]
```diff
@@ -89,21 +91,114 @@ public void setUp()
...
+                .addRoundTrip("smallint", "32767", SMALLINT, "SMALLINT '32767'") // max value in SQL Server and Trino
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_smallint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_smallint"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_smallint"));
+    }
+
+    @Test
+    public void testUnsupportedSmallint()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase()::execute, "test_unsupported_smallint", "(data smallint)")) {
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:142]:
```suggestion
        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_smallint", "(data smallint)")) {
```

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 8]
```diff
@@ -89,21 +91,114 @@ public void setUp()
...
+                .addRoundTrip("integer", "2147483647", INTEGER, "2147483647") // max value in SQL Server and Trino
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_int"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_int"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_int"));
+    }
+
+    @Test
+    public void testUnsupportedInteger()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase()::execute, "test_unsupported_integer", "(data integer)")) {
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:168]:
```suggestion
        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_integer", "(data integer)")) {
```

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java [hunk 9]
```diff
@@ -89,21 +91,114 @@ public void setUp()
...
+                .addRoundTrip("bigint", "9223372036854775807", BIGINT, "9223372036854775807") // max value in SQL Server and Trino
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_bigint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_bigint"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_bigint"));
+    }
+
+    @Test
+    public void testUnsupportedBigint()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase()::execute, "test_unsupported_bigint", "(data bigint)")) {
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:194]:
```suggestion
        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_bigint", "(data bigint)")) {
```

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java (hunk 7 above)
Comment (tangjiangling) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java:142]: My stupidity!

**Comment by tangjiangling** [comment]
Updated.

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**